### PR TITLE
Post Schema Initial Setup

### PR DESCRIPTION
**Pull request by christinem**
Set up initial basic tables for Posts and Post Categories. Also set up initial parsing overhead to work when running `stack exec courseography database`. At the current moment, all it does is run two function calls, one that inserts a new post and one that inserts a new post category. Tested it by commenting out all function calls in `parseAll` except for `parsePosts` (since the rest of the parsing is currently broken with the new timetable format), and ran the above stack command, and can confirm it properly creates the new tables and adds the new values.

### CODE REVIEW

File: app/Database/PostInsertion.hs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import Config (databasePath)
+import Database.Persist.Sqlite (selectFirst, fromSqlKey, toSqlKey, insertMany_, insert_, insert, SqlBackend, (=.), (==.), updateWhere, runSqlite, runMigration)
+import Database.Tables
+import Data.Aeson
+
+-- | Insert a new post into the database
+insertPost :: T.Text -> T.Text -> T.Text -> IO ()
+insertPost departmentName postName postCode =
+    runSqlite databasePath $ do
+        runMigration migrateAll
```
Comment (david-yz-liu) [inline app/Database/PostInsertion.hs]: You don't need to run the migration every time; you can remove this line.

File: app/WebParsing/PostParser.hs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE OverloadedStrings #-}
+module WebParsing.PostParser
+    (parsePosts) where
+
+import Network.HTTP
```
Comment (david-yz-liu) [inline app/WebParsing/PostParser.hs]: Look carefully at what dependencies are actually needed. It's a good practice to start from scratch and just add the dependencies as you need them. (So remove the ones that aren't necessary.)

**Comment by david-yz-liu** [comment]
@christinem I'd like to merge this in, if you have time to clean it up a little first. The basic functionality is acceptable.

**Comment by christinem** [comment]
@david-yz-liu sorry for the lateness, but I've removed the unnecessary inputs and the unnecessary migration line

**Pull request merged**